hnn_core/tests/test_network.py

```diff
@@ -855,9 +856,25 @@ def test_tonic_biases():
     assert 'tonic' in net.external_biases
     assert 'L5_pyramidal' not in net.external_biases['tonic']
     assert net.external_biases['tonic']['L2_pyramidal']['t0'] == 0
-    with pytest.raises(ValueError, match=r'Tonic bias already defined for.*$'):
+    with pytest.raises(ValueError, match=r'Bias named tonic already defined '
+                       r'for.*$'):
         net.add_tonic_bias(amplitude=tonic_bias_2)
 
+    # non-existent section
+    net.external_biases = dict()
+
+    with pytest.raises(ValueError, match=r'section must be an existing '
+                       r'section of the current cell or None. '
+                       r'Got apical_4.'):
+        net.add_tonic_bias(amplitude={'L2_pyramidal': .5}, section='apical_4')
```

Comment:
you should add a test that checks in `net.external_biases` once a bias has been added:

```py
assert 'blah' in net.external_biases
assert net.external_biases['blah']['t0'] == 100
```

etc